src/anim/evaluator/anim-clip.js

```diff
@@ -83,13 +89,52 @@ class AnimClip {
         this._blendOrder = blendOrder;
     }
 
+    get eventCursor() {
+        return this._eventCursor;
+    }
+
+    set eventCursor(value) {
+        this._eventCursor = value;
+    }
+
+    activeEventsForFrame(frameStartTime, frameEndTime) {
+        if (frameStartTime === 0) {
+            this.eventCursor = 0;
+        }
+        var events = [];
+        var clippedFrameDuration;
+        // if this frame overlaps with the end of the track, we should clip off the end of the frame time then check that clipped time later
+        if (frameEndTime > this.track.duration) {
+            clippedFrameDuration = frameEndTime - this.track.duration;
```

Comment:
this should handle a case when the time wraps around more than once .. which it can for short animations and long dt. I think this should have a unit test written for all these edge cases.

Reply:
This function is called recursively until all of the current frames duration is consumed.

Reply:
ah yeah I missed that .. that seems ok